pkg/kubelet/lifecycle/handlers.go

```diff
@@ -123,8 +124,17 @@ func (hr *HandlerRunner) runHTTPHandler(pod *v1.Pod, container *v1.Container, ha
 			return "", err
 		}
 	}
-	url := fmt.Sprintf("http://%s/%s", net.JoinHostPort(host, strconv.Itoa(port)), handler.HTTPGet.Path)
-	resp, err := hr.httpGetter.Get(url)
+	path := handler.HTTPGet.Path
+	u, err := url.Parse(path)
+	// Something is busted with the path, but it's too late to reject it. Pass it along as is.
+	if err != nil {
+		u = &url.URL{
+			Path: path,
+		}
+	}
+	u.Scheme = string(handler.HTTPGet.Scheme)
```

Comment:
The scheme is an enum API value, switch on the values and set to "http" or "https", don't use it directly in the url

Reply:
Also, consider defaulting to http when the scheme is unspecified